File: Java/graphs/Prim.java
package graphs;

/**
 * Created by cristian on 7/4/2015.
 */
public class Prim {

    static int N = 5;
    static int MAX_VALUE = 999;


    public static void main(String[] args) {
        int[][] matrix = buildMatrixGraph();
        primSpanningTree(matrix, 1);
    }

    private static void primSpanningTree(int[][] matrix, int startingPosition) {
        int[] visited = new int[N];
        int startNode = 0, endNode = 0;
        int minimum = MAX_VALUE;
        int total = 0;
        int noOfEdges = 1;
        visited[startingPosition] = 1;
        while (noOfEdges < N) {
            minimum = MAX_VALUE;
            for (int row = 0; row < N; row++) {
                if (visited[row] != 1) {
                    for (int column = 0; column < N; column++) {
                        if (visited[column] != 1 && minimum > matrix[row][column]) {
                            minimum = matrix[row][column];
                            startNode = row;
                            endNode = column;
                        }
                    }
                }
            }
            visited[endNode] = 1;
            System.out.println("Edge found: " + (startNode + 1) + "--> " + (endNode + 1) + " with the minimum the cost of: " + minimum);
            noOfEdges++;
            total += minimum;
        }
        System.out.println("Weight of the minimum spanning tree is : " + total);
    }


    private static int[][] buildMatrixGraph() {
        return new int[][]{
        /* first node*/ {MAX_VALUE, 100, 110, MAX_VALUE, 50},
        /* second node*/{100, MAX_VALUE, 15, 40, MAX_VALUE},
        /* third node*/ {110, 15, MAX_VALUE, 70, MAX_VALUE},
        /* fourth node*/{MAX_VALUE, 40, 70, MAX_VALUE, 16},
        /* fifth node*/ {50, MAX_VALUE, MAX_VALUE, 16, MAX_VALUE}
        };
    }
}

File: Exam/prim.c
#include <stdlib.h>
#include <stdio.h>
#include "prim.h"




void primSpanningTree(int matrix[5][5], int startingPosition, int numberOfNodes){
	int MAX_VALUE = 999;
	int *visited = (int*)malloc(numberOfNodes * sizeof(int));
	int startNode, endNode, min, row, column, noOfEdges, total;
	total = 0;
	noOfEdges = 1;
	min = MAX_VALUE;
	for (row = 0 ; row < numberOfNodes; row++){
		visited[row] = 0;
	};
	visited[startingPosition] = 1;
	while (noOfEdges < numberOfNodes){
			min = MAX_VALUE;
		for (row = 0; row < numberOfNodes; row++){
			if (visited[row] == 0){
				for (column = 0; column < numberOfNodes; column++){
					if (visited[column]==0 && matrix[row][column] < min){
						startNode = row;
						endNode = column;
						min = matrix[row][column];
					}
				}

		};
	
	}

			visited[endNode] = 1;
		printf("Found edge from %d to %d with cost %d \n", (startNode + 1), (endNode + 1), min);
		total+=min;
		noOfEdges++;
	}
	printf("Shortest path is %d\n", total);
}